File: GenZCoders-main/Frontend/src/api/models/application.ts
export type ApplicationStatusId = number;

export type ApplicationDto = {
  id: number | string;
  courseRoundId: number;
  accountId?: number;
  applicationDate?: string;
  status?: string;
  fullNameEn?: string;
  email?: string;
  phone?: string;
  answer1?: string | null;
  answer2?: string | null;
  answer3?: string | null;
  answer4?: string | null;
  answer5?: string | null;
  answer6?: string | null;
  answer7?: string | null;
  answer8?: string | null;
  answer9?: string | null;
  answer10?: string | null;
  statusId?: ApplicationStatusId;
  studentId?: string;
  appliedAt?: string;
  reviewedAt?: string;
  reviewedBy?: string;
  notes?: string;
  /** Payment receipt uploaded by the student (set once they upload it). */
  media?: { filePath?: string | null } | null;
};

export type CreateApplicationRequest = {
  courseRoundId: number;
  answer1?: string | null;
  answer2?: string | null;
  answer3?: string | null;
  answer4?: string | null;
  answer5?: string | null;
  answer6?: string | null;
  answer7?: string | null;
  answer8?: string | null;
  answer9?: string | null;
  answer10?: string | null;
};

export type UpdateApplicationStatusRequest = {
  statusId: ApplicationStatusId;
};

export type UpdateApplicationCourseRoundRequest = {
  courseRoundId: number;
};

File: GenZCoders-main/Backend/GenZCoders/Models/Application.cs
using System;
using System.Collections.Generic;

namespace GenZCoders.Models;

public partial class Application
{
    public long Id { get; set; }

    public long CourseRoundId { get; set; }
    public long AccountId { get; set; }

    public DateTime ApplicationDate { get; set; }

    public long StatusId { get; set; }

    public string? Answer1 { get; set; }

    public string? Answer2 { get; set; }

    public string? Answer3 { get; set; }

    public string? Answer4 { get; set; }

    public string? Answer5 { get; set; }

    public string? Answer6 { get; set; }

    public string? Answer7 { get; set; }

    public string? Answer8 { get; set; }

    public string? Answer9 { get; set; }

    public string? Answer10 { get; set; }
    public virtual Account Account { get; set; }
    public virtual Status Status { get; set; }
    public virtual CourseRound CourseRound { get; set; }

}
